### ISSUE TITLE
Automation tab in VM creation page should mention that it relies on qemu-guest-agent being installed in the guest system

### ISSUE THREAD

**Issue by travier** [cockpit-project/cockpit-machines#1265]
While trying to understand how https://github.com/cockpit-project/cockpit-machines/pull/1150 is effectively implemented, I've realized that this relies on https://wiki.qemu.org/Features/GuestAgent and https://wiki.libvirt.org/Qemu_guest_agent.html and thus needs the qemu-guest-agent to be installed and running in the Guest system.

One notable system that does not include the QGA is Fedora CoreOS (see https://github.com/coreos/fedora-coreos-tracker/issues/74).

The suggestion is thus to include a notice somewhere on this tab that this relies on the agent to be there in the guest.

**Comment by mvollmer** [comment]
> While trying to understand how [#1150](https://github.com/cockpit-project/cockpit-machines/pull/1150) is effectively implemented, I've realized that this relies on https://wiki.qemu.org/Features/GuestAgent and https://wiki.libvirt.org/Qemu_guest_agent.html and thus needs the qemu-guest-agent to be installed and running in the Guest system.

I don't think this is true. The automation is done via cloud-init, specifcally the `--cloud-init` option of `virt-install`. The guest agent is not involved. Indeed, the guest agent protocol has no means to install SSH keys into the guest.

**Comment by travier** [comment]
The QEMU Guest Agent can write arbitrary files to the VM filesystem: https://qemu-project.gitlab.io/qemu/interop/qemu-ga-ref.html#command-QGA-qapi-schema.guest-file-write

But I indeed missed that they are passed as part of a cloud-init config.

**Comment by travier** [comment]
Looks like what I would need is https://github.com/cockpit-project/cockpit-machines/issues/1835 instead then.

**Comment by mvollmer** [comment]
> The QEMU Guest Agent can write arbitrary files to the VM filesystem

OMG, it can! Thanks, I didn't know that. I am surprised, tbh.

**Comment by mvollmer** [comment]
> Looks like what I would need is [#1835](https://github.com/cockpit-project/cockpit-machines/issues/1835) instead then.

Yes, I think so, too.

Do you have any idea how Cockpit can know whether to use cloud-init or ignition for a OS installation?

**Comment by travier** [comment]
> > Looks like what I would need is [#1835](https://github.com/cockpit-project/cockpit-machines/issues/1835) instead then.
> 
> Yes, I think so, too.
> 
> Do you have any idea how Cockpit can know whether to use cloud-init or ignition for a OS installation?

There is no way to know from the outside unfortunately. We have to let the user choose the option that is used by the image they are picking.

**Comment by mvollmer** [comment]
> > > Looks like what I would need is [#1835](https://github.com/cockpit-project/cockpit-machines/issues/1835) instead then.
> > 
> > 
> > Yes, I think so, too.
> > Do you have any idea how Cockpit can know whether to use cloud-init or ignition for a OS installation?
> 
> There is no way to know from the outside unfortunately. We have to let the user choose the option that is used by the image they are picking.

We could hope that libosinfo provides this information, but it is probably not a good idea to rely on it. So some UI might indeed be necessary.
